tests/testthat/_snaps/highlight.md

```diff
@@ -0,0 +1,19 @@
+# pre() can produce needed range of outputs
+
+    Code
+      cat(pre("x"))
+    Output
+      <pre>x</pre>
+    Code
+      cat(pre("x", r_code = TRUE))
+    Output
+      <pre><code class='sourceCode R'>x</code></pre>
```

Comment:
Note that for some reason Pandoc would add the sourceCode class on the div, pre and code. I am however not sure it needs to be imitated 

``` r
text <- paste("```r", "1 + 1", "```", sep = "\n")
file <- withr::local_tempfile()
#> Setting deferred event(s) on global environment.
#>   * Execute (and clear) with `withr::deferred_run()`.
#>   * Clear (without executing) with `withr::deferred_clear()`.
writeLines(text, file)
rmarkdown::pandoc_convert(file, "html", "markdown", fs::path_ext_set(file, ".html"))
readLines(fs::path_ext_set(file, ".html"))
#> [1] "<div class=\"sourceCode\" id=\"cb1\"><pre class=\"sourceCode r\"><code class=\"sourceCode r\"><span id=\"cb1-1\"><a href=\"#cb1-1\" aria-hidden=\"true\" tabindex=\"-1\"></a><span class=\"dv\">1</span> <span class=\"sc\">+</span> <span class=\"dv\">1</span></span></code></pre></div>"
```

<sup>Created on 2021-05-07 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0.9000)</sup>

Reply:
Just for reference, this is what pandoc generates:

```html
<div class="sourceCode" id="cb1">
  <pre class="sourceCode r">
    <code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="dv">1</span> <span class="sc">+</span> <span class="dv">1</span></span></code>
  </pre>
</div>
```

So while the `<div>` has `sourceCode` only the `<pre>` and `<code>` have the language. But maybe that's ok — we only need the language for downlit, and downlit is done by the time we get to the relevant code in pkgdown.

Reply:
Looking at the downlit code again, I'm now leaning towards producing something like this:

```html
<div class="sourceCode">
  <pre class="sourceCode r"><code>
    ...
  </code></pre>
  <pre class="sourceCode r-out"><code>
    ...
  </code></pre>
</div>
      ```

The main difference between out syntax highlighting and pandoc's is that there might be multiple `<pre>`s inside a `div.sourceCode`